likelihood/sacc/sacc_likelihoods/twopoint.py

```diff
@@ -6,100 +6,112 @@
+    category = kwargs.get("category")
+    if category == "spectrum":
+        x_theory = block[section, "ell"]
+    elif category == "real":
+        x_theory = block[section, "theta"]
+    elif category == "cosebi":
+        x_theory = block[section, "cosebis_n"]
+    is_auto = block[section, "is_auto"]
+
+    # We build up these vectors from all the data points.
+    # Only the theory vector is needed for the likelihood - the others
+    # are for convenience, debugging, etc.
+    theory_vector = []
+    angle_vector = []
+    bin1_vector = []
+    bin2_vector = []
+
+
+    # Because we called to_canonical_order when we loaded the data,
+    # we know that the data is grouped by data type, and then by tracers (tomo bins).
+    # So that means we can do a data type at a time and then concatenate them, and
+    # within this do a bin pair at a time, and concatenate them too.
+    for b1, b2 in sacc_data.get_tracer_combinations(data_type):
+        # Here we assume that the bin names are formatted such that
+        # they always end with _1, _2, etc. That isn't always true in
+        # sacc, but is somewhat baked into cosmosis in other modules.
+        # It would be nice to update that throughout, but that will
+        # have to wait. Also, cosmosis bins start from 1 not 0.
+        # We need to make sure that's fixed in the 
+        i = int(b1.split("_")[-1]) + 1
+        j = int(b2.split("_")[-1]) + 1
+
+        if data_type in kwargs.get("flip", False):
+            i, j = j, i
+
+        try:
+            theory = block[section, f"bin_{i}_{j}"]
+        except BlockError:
+            if is_auto:
+                theory = block[section, f"bin_{j}_{i}"]
+            else:
+                raise
+
+        # check that all the data points share the same window
+        # object (window objects contain weights for a set of ell / theta values,
+        # as a matrix), or that none have windows.
+        window = None
+        for d in sacc_data.get_data_points(data_type, (b1, b2)):
+            w = d.get_tag('window')
+            if (window is not None) and (w is not window):
+                raise ValueError("Sacc likelihood currently assumes data types share a window object")
+            window = w
+
+        # We need to interpolate between the sample ell / theta values
+        # onto all the ell / theta values required by the weight function
+        # This will give zero outside the range where we have
+        # calculated the theory
+        theory_spline = SpectrumInterp(x_theory, theory)
+        if window is not None:
+            x_window = window.values
+            theory_interpolated = theory_spline(x_window)
+
+        for d in sacc_data.get_data_points(data_type, (b1, b2)):
+            if category == "spectrum":
+                x_nominal = d['ell']
+            elif category == "real":
+                x_nominal = d['theta']
+            #TO-DO: Not sure if this works for COSEBIs, come back to it later.
```

Comment:
What's the status of this?  If it might not work yet then it's good to raise NotImplementedError.